python/openassetio/test/managerValidator/commandLine.py

```diff
@@ -0,0 +1,49 @@
+"""
+Utilities for executing the manager test harness from the command line.
+"""
+import argparse
+import collections
+import importlib.util
+import unittest
+
+
+def execute(argv, parser, testModule, harnessFactory):
+    parsedArgs = parser.parse(argv)
+    logger = harnessFactory.createLogger()
+    hostInterface = harnessFactory.createHostInterface()
+    managerFactory = harnessFactory.createManagerFactory(logger)
+    session = harnessFactory.createSession(
+        parsedArgs.fixtures["identifier"], hostInterface, logger, managerFactory)
+    unittestLoader = harnessFactory.createLoader(parsedArgs.fixtures, session)
+    validatorHarness = harnessFactory.createHarness(unittest.main, unittestLoader)
+    return validatorHarness.executeTests(parsedArgs.extraArgs, testModule)
+
+
+class Parser:
+    _Parsed = collections.namedtuple("Parsed", ("fixtures", "extraArgs"))
+
+    def __init__(self, fixtureLoader):
+        self.__fixtureLoader = fixtureLoader
+
+    def parse(self, argv):
+        cmdline = argparse.ArgumentParser(
+            prog="managerValidator", description="Execute OAIO manager test harness")
+        cmdline.add_argument(
+            "-f", "--fixtures", metavar="FILE", required=True, help="Path to fixture file")
+        cmdline.add_argument(
+            "extraArgs", metavar="ARGS", nargs=argparse.REMAINDER,
+            help="Arguments to pass to the test runner")
+
+        args = cmdline.parse_args(argv[1:])
+
+        return Parser._Parsed(
```

Comment:
Interesting point, not something I've thought about. In this case it's a bit in-betweeny since it's returning a `namedtuple`, which you could argue is just syntactic sugar on a tuple. 

I guess the question is, if someone sees an `_` prefix, do they assume only that the class is never instantiated outside its scope, or assume more broadly that instances are never even seen outside of that scope.  Presumably you're in the second camp.  I'm happy to go with that.

Reply:
I think the second has been our rule of thumb in the past, as it works in languages where visibility is respected too.

Reply:
OK I'm happy to go with that.  For info, [C++ doesn't respect visibility in this way](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:19,positionColumn:2,positionLineNumber:19,selectionStartColumn:2,selectionStartLineNumber:19,startColumn:2,startLineNumber:19),source:'%0Aclass+Public%0A%7B%0Aprivate:%0A++++struct+Private%0A++++%7B%0A++++++++int+data%3B%0A++++%7D%3B%0Apublic:%0A++++Private+getPrivateInstance(int+data)%0A++++%7B%0A++++++++return+Private%7Bdata%7D%3B%0A++++%7D%0A%7D%3B%0A%0Aint+main(int+argc,+char**+argv)%0A%7B%0A++++return+Public%7B%7D.getPrivateInstance(123).data%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:41.421457673155274,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:clang700,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'1',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!((name:boost,ver:'170')),options:'-std%3Dc%2B%2B14',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1,tree:'1'),l:'5',n:'0',o:'x86-64+clang+7.0.0+(C%2B%2B,+Editor+%231,+Compiler+%231)',t:'0')),k:43.26072157930565,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compiler:1,editor:1,fontScale:14,fontUsePx:'0',wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+clang+7.0.0+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:58.57854232684472,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

Reply:
Today we learnt: https://stackoverflow.com/questions/13532784/why-can-i-use-auto-on-a-private-type!